### ISSUE TITLE
[MOD] Notifications panel: fetch and load content using async

### ISSUE THREAD

**Issue by aclist** [aclist/kbin-kes#133]
Current panel implementation uses an iframe to force the entire notifications page of the site into the view.

Proposal:

Fetch the names/messages elements of that page in async, use a progress animation in the panel itself, then load in the contents when ready. Also gives more fine-grained control over how the elements are arranged and manipulated, as they could just be stored in arrays rather than rendering the whole page.

**Comment by aclist** [comment]
@artillect @minnieoo End-to-end prototype is ready in this branch: https://github.com/aclist/kbin-kes/tree/patch-notipanel

Content is fetched async and stored in arrays. A spinner animates while loading. When ready, the content loads in.

The code is a lot more compact now and takes a more function => callback oriented structure.

Now we need help from the CSS masters: currently no styling is being applied to the messages in the panel. I think a grid template would probably do the trick. We can lay them out in the same way as the old CSS and give them the same class names if necessary--so that it looks identical, but we're actually loading the content from arrays instead of an iframe.

The "Read" and "Purge" buttons will be easy to add, they are just a POST request to the server.

**Comment by aclist** [comment]
Also, the spinner animation could be anything, I just threw that one in as an example. Could center it in the panel or do whatever with it, too.

**Comment by aclist** [comment]
- [x] Add time element after name
- [x] Add unique classname to private message
- [ ] Add read/unread classnames -- pending

Completed and will commit those updates later after testing it.

**Comment by honkaiimpact3rd** [comment]
I doubt pagination would work well in a panel from a user perspective, my idea so far is to have infinite scroll or a "show more" button at the bottom that loads the next page of notifications (make it a preference or make it depend on the "infinite scroll" setting in kbin settings? maybe?)

I never had multiple pages of notifications (I purge them regularly) so I have no clue how it really looks like though lol

**Comment by aclist** [comment]
> Infinite scroll

Hmm, that's not a bad idea. I think you could safely fit a small arrow-left/arrow-right icon at the top or bottom too if necessary.

I never had thousands of notifications, either, so don't have any real-world data on that, but the first page tends to hold quite a lot of notifications before you need to reach page 2.

**Comment by aclist** [comment]
Took a look at the unread/read, unread messages just come through with a reduced opacity class:

Read: `section section--small thread`
Unread: `section section--small thread opacity-50`
```
.opacity-50 {
    opacity: 0.5
}
```

So we can just carry that classname through with no change and should be good!

**Comment by aclist** [comment]
Alright, added in everything you asked for up to commit 6f7e229

- PMs get the `noti-message` class
- Read items pass through `opacity-50`
- Time element comes after username

Pagination: to be determined

**Comment by honkaiimpact3rd** [comment]
Great! There are currently 2 issues with the styling I made:
* Styling for the notifications counter doesn't use the child combinator (it wasn't needed before so I just didn't put it) so it affects everything in the panel. It can be easily fixed by changing `#header menu .notification-button a:not(.fa-solid.fa-bell)` to `#header menu .notification-button > a:not(.fa-solid.fa-bell)`
* Kbin's styling for `.dropdown__menu a, .dropdown__menu button` has `!important` for `padding` and `display` _(absolutely no clue why, deleting it changes nothing lol)_, which breaks styling inside the panel, I tried to add `!important` to my own styling in Stylus to override but it wouldn't work (I delete it manually while working on it), might have to either
  * Make an issue on Kbin to fix it, might take a while as it would be a low priority (but easy to fix) issue.
  * Use element styling, it's already used to add `margin-left` anyway. 

Another slight nitpick is with the counters. Notifications and message counters are separate _(despite both being displayed on the notifications page for some reason?)_, currently it moves the notifications counter to the notification bell but the messages counter is untouched, it looks a bit ugly in my opinion, but we can probably keep it since clicking the messages counter directly takes you to the messages page and changing it will add an extra click and second which is something that's the mod is supposed to stop lol
